sklearn/cluster/k_means_.py

```diff
@@ -441,10 +441,19 @@ def _kmeans_single(X, n_clusters, x_squared_norms, max_iter=300,
             best_centers = centers.copy()
             best_inertia = inertia
 
-        if squared_norm(centers_old - centers) <= tol:
+        shift = squared_norm(centers_old - centers)
+        if shift <= tol:
             if verbose:
                 print("Converged at iteration %d" % i)
+
             break
+
```

Comment:
Can you add a code comment here; something like ``# rerun E-step in case of non-convergence so that predicted labels match cluster centers``